49-hvac-current-monitor/firmware/src/main.cpp

```diff
@@ -0,0 +1,330 @@
+#include <Notecard.h>
+#include "NotecardEnvVarManager.h"
+
+// Uncomment this line and replace com.my-company.my-name:my-project with your
+// ProductUID.
+// #define PRODUCT_UID "com.my-company.my-name:my-project"
+
+#ifndef PRODUCT_UID
+#define PRODUCT_UID ""
+#pragma message "PRODUCT_UID is not defined in this example. Please ensure your Notecard has a product identifier set before running this example or define it in code here. More details at https://bit.ly/product-uid"
+#endif
+
+#ifndef OUTBOUND_MINS
+#define OUTBOUND_MINS 10
+#endif
+
+#ifndef INBOUND_MINS
+#define INBOUND_MINS 5
+#endif
+
+#ifndef MAX_SERIAL_WAIT_MS
+#define MAX_SERIAL_WAIT_MS 5000
+#endif
+
+#ifndef ENV_FETCH_INTERVAL_MS
+#define ENV_FETCH_INTERVAL_MS (3 * 60 * 1000) // 3 minutes
+#endif
+
+#ifndef ANALOG_PIN
+#define ANALOG_PIN A0
+#endif
+
+#ifndef ADC_BITS
+#define ADC_BITS 10
+#endif
+
+#define ADC_RANGE (1 << ADC_BITS)
+#define DEFAULT_ADC_OFFSET (ADC_RANGE >> 1)
+
+#ifndef SAMPLES_PER_CALC
+#define SAMPLES_PER_CALC 1024
+#endif
+
+#ifndef DEFAULT_SECONDS_BETWEEN_MEASUREMENTS
+#define DEFAULT_SECONDS_BETWEEN_MEASUREMENTS 15
+#endif
+
+#ifndef DEFAULT_PCT_CHANGE_THRESHOLD
+#define DEFAULT_PCT_CHANGE_THRESHOLD 50
+#endif
+
+#ifndef AMPS_PER_VOLT
+#define AMPS_PER_VOLT 30
+#endif
+
+#ifndef ADC_REF_VOLTAGE
+#define ADC_REF_VOLTAGE 3.3
+#endif
+
+#ifndef MEASUREMENTS_PER_NOTE
+#define MEASUREMENTS_PER_NOTE 30
+#endif
+
+#ifndef MEASUREMENTS_FILE
+#define MEASUREMENTS_FILE "measurements.qo"
+#endif
+
+Notecard notecard;
+NotecardEnvVarManager *envVarManager = NULL;
+
+// These are the environment variables we'll be fetching from the Notecard.
+static const char *envVars[] = {
+    "seconds_between_measurements",
+    "percentage_change_threshold"
+};
+static const size_t numEnvVars = sizeof(envVars) / sizeof(envVars[0]);
+static int msBetweenMeasurements = DEFAULT_SECONDS_BETWEEN_MEASUREMENTS * 1000;
+static float pctChangeThreshold = DEFAULT_PCT_CHANGE_THRESHOLD;
+
+static float measurements[MEASUREMENTS_PER_NOTE];
+static size_t measurementsIdx = 0;
+
+static unsigned long lastMeasurementMs = 0;
+static unsigned long lastEnvFetchMs = 0;
+static float lastAvgCurrent = -1;
+
+void sendMeasurementsNote(const float *data, size_t len,
```

Comment:
A guard here for when there are 0 measurements will help avoid false change reports when the average becomes 0 because this happened after measurements were sent and then immediately afterwards environment variables updated.

Reply:
Good suggestion.